### DISCUSSION TITLE
Lesson 1 - Gas fees. 2 questions

### DISCUSSION THREAD

**Discussion by kidoMiike** [Cyfrin/foundry-full-course-cu#98] [Q&A] [answered]
Hope you can help...
I didn't understand 2 things.
1. If gas is the amount of computations used, then it should be something that is calculated by transaction or contract. How come you can choose the amount of gas you want to spend in your wallet? 
2. How does the wallet calculate the recommended priority fee?
Thanks for answers!

**Comment by cromewar** [comment, marked as answer]
Hello there @kidoMiike 
1. The reasons is because you have a minimum gas fee, which is the the minimum fee required to pay the nodes to write the transaction (validators on PoS), so you can select the amount you want to pay, if you pay more your transaction will have priority so the transaction will go trough faster, but if you pay the bare minimum and the transaction fee increases your transaction have the risk to get reverted.
2. They uses libraries such as web3 or APIs such as ETHGasStation, depending on the wallet. So they take the minimum gas fee required and allow you to select a range, usually SLOW, STANDARD and FAST. they use that as an approximation other users are paying and for example if you select fast, you will pay more yes, but they try to calculate the threshold on which paying more will not make your transaction faster so you pay more but not in exaggeration.


I hope you find this answer useful, please let us know if you have more questions.

**Reply by kidoMiike** [reply]
Thanks alot :))
